Added EiffelSourceChangeCreatedEvent documentation.

d-stahl-ericsson opened the pull request.

p-backman-ericsson commented on eiffel-vocabulary/EiffelSourceChangeCreatedEvent.md:
```diff
@@ -0,0 +1,210 @@
+__Required:__ Yes  
+__Description:__ The branch of the change.
+
+#### data.ccCompositeIdentifier.configSpecPath
```
Shouldn't we use a URI for the configSpecPath?

d-stahl-ericsson replied:
I think we should.

Andrey9kin commented on eiffel-vocabulary/EiffelSourceChangeCreatedEvent.md:
```diff
@@ -0,0 +1,210 @@
+#### data.issues.type
+__Type:__ String  
+__Required:__ Yes  
+__Legal values:__ BUG, IMPROVEMENT, FEATURE, WORK_ITEM, REQUIREMENT, OTHER  
```
What is the difference between BUG and WORK_ITEM? Or FEATURE and WORK_ITEM? They all look like work items to me

d-stahl-ericsson replied:
BUG or FEATURE describe the product and are long lived artifacts. WORK_ITEM is something you put in your backlog, describing work to be done (which could be "Fix that bug over there!"). Some people like linking one, some like linking the other, and some like linking both.

Andrey9kin replied:
But how come something that have to be done have anything to do with the model that describes events that already happened? Aren't we mixing things?

d-stahl-ericsson replied:
I disagree. The event says that something happened: sources were changed. Why did this happen? Because of a work item, saying this had to be done. At the time you look at the event the work item may be open or closed or anything in between, but that's besides the point. You still want to be able to trace changes you made (and thereby product deltas) to work items.

d-stahl-ericsson commented on eiffel-vocabulary/EiffelSourceChangeCreatedEvent.md:
```diff
@@ -0,0 +1,210 @@
+__Required:__ Yes  
+__Description:__ The URI of the repo.
+
+### data.gerritIdentifier
```
It's a balance, I guess. As you say, you could go conservative and treat it as any other Git repo, or you could make allowances for particular information (e.g. patch set, Change-Id) where convenient. As a rule I favor conservatism, so I think you have a point.

Andrey9kin commented on eiffel-vocabulary/EiffelSourceChangeCreatedEvent.md:
```diff
@@ -0,0 +1,210 @@
+__Description:__ The email address of the author.
+
+#### data.author.id
+__Type:__ String  
```
I would suggest removing id and group - there is no good generic way of identifying those automatically. I believe that we should strive to have the data model that allows automated created of the events and use it as a guiding principal. I mean, every time we want to add something we should think about the end users - how hard it will be to implement automated gathering of such info. What do you think?

d-stahl-ericsson replied:
I think it's fine to include non-mandatory fields which may or may not be relevant in each particular case, the principle being "You're free to decide what to say, but if you want to say it, here's how".

Andrey9kin commented on eiffel-vocabulary/EiffelSourceChangeCreatedEvent.md:
```diff
@@ -0,0 +1,210 @@
+__Required:__ Yes  
+__Description:__ The URI of the repository containing the change.
+
+### data.ccCompositeIdentifier
```
I guess this covers base ClearCase, right? Shall we have something for UCM? Or is it possible to describe UCM coordinates using fields below?

d-stahl-ericsson replied:
I'm not familiar enough with UCM to answer that. This syntax is based on historical usage - whether that usage is UCM or base ClearClase I don't know.

Andrey9kin commented on eiffel-vocabulary/EiffelSourceChangeCreatedEvent.md:
```diff
@@ -0,0 +1,210 @@
+__Description:__ Identifier of a composite ClearCase change – in other words, not single file commit, but analogous of repository-wide commits of e.g. SVN or Git.
+
+#### data.ccCompositeIdentifier.vob
+__Type:__ String  
```
Should it be a list? Change may affect multiple vobs

d-stahl-ericsson replied:
Doesn't that fall under the same reasoning as changes to multiple repositories? I.e. send multiple events if you change multiple repositories.

d-stahl-ericsson replied:
Having bounced this off of a few people and given the fact that a configSpec can and often does deal with multiple VOBs, I'll push a commit changing this to a list.

d-stahl-ericsson commented:
I removed the Gerrit identifier and added details, tracker and id as optional fields to data.change instead. This should afford a more generic approach: the basic identifier is the same, but if you're using supporting tools you can reference them there in addition to the identifier.

I'm bouncing the ClearCase identifier off of some people to get their feedback.

p-backman-ericsson commented:
:+1:

danielyinanc commented:
:+1: 

This event type, like the EnvironmentDefined event could end up being a very flexible event representing real life more than other more CI-ish events. I propose we live with what we have and modify as needed.
